lib/types/rules/best-practices.d.ts

```diff
@@ -692,6 +692,11 @@ export interface BestPractices extends Linter.RulesRecord {
                  * @default []
                  */
                 ignorePropertyModificationsFor: string[];
+                /**
+                 * @since 6.6.0
+                 * @default []
+                 */
+                ignorePropertyModificationsForRegex: string[];
```

Comment:
Both of these properties are only valid when `props` is `true`, right now I think `{ props: false,  ignorePropertyModificationsFor: ["bar"]}` will be valid according to types.


https://github.com/eslint/eslint/blob/c153084250673b31bed46e3fe6af7a65b4ce8d6f/lib/rules/no-param-reassign.js#L26-L59

Reply:
Good call, I’ve just improved the types accordingly.

Thanks for the feedback 🙌